torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -152,7 +152,6 @@ static ScalarType dtype_to_aten(int dtype) {
     case NPY_DOUBLE: return kDouble;
     case NPY_FLOAT: return kFloat;
     case NPY_HALF: return kHalf;
-    case NPY_LONGLONG:
```

Comment:
MSVC doesn't support two switch cases (`NPY_LONGLONG` and `NPY_INT64`) that have the same value

Reply:
Removing that case altogether isn't a good fix. Just repeat the body

Reply:
Doing
```
case NPY_LONGLONG: return kLong; 
case NPY_INT64: return kLong; 
```
seems to still have MSVC complaining about `error C2196: case value '9' already used`.

I am thinking of a workaround such as:
```
switch (dtype) {
    case NPY_DOUBLE: return kDouble;
    case NPY_FLOAT: return kFloat;
    case NPY_HALF: return kHalf;
    case NPY_INT32: return kInt;
    case NPY_INT16: return kShort;
    case NPY_UINT8: return kByte;
    default:
      // Workaround: MSVC does not support two switch cases that have the same value
      if (dtype == NPY_LONGLONG || dtype == NPY_INT64) {
        return kLong;
      } else {
        break;
      }
  }
```

Reply:
What about using macros like 
```
case NPY_INT64:
#if NPY_LONGLONG != NPY_INT64
case NPY_LONGLONG:
#endif
```?

Reply:
If using macro as suggested, Linux test seems to still give this error:
```
======================================================================
ERROR: test_from_numpy (__main__.TestTorch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_torch.py", line 4935, in test_from_numpy
    tensor_from_array = torch.from_numpy(array)
TypeError: can't convert np.ndarray of type numpy.int64. The only supported types are: double, float, float16, int64, int32, and uint8.

----------------------------------------------------------------------
```

Reply:
Eh ok, let's go with the default, but you should mention that the error happens because for MSVC `NPY_LONGLONG` and `NPY_INT64` are the same thing